File: Stack/3_seek.cpp
#include <iostream>
using namespace std;

class Stack
{
    int size;
    int top;
    int *data;

    public:
    Stack(int size)
    {
        this->size = size;
        this->top  = -1;
        this->data  = new int[size];
    }

    void push(int element)
    {
        if(isFull())
        {
            cout<<"Can not push to already full stack";
            return;
        }

        top++;
        this->data[top] = element;
    }

    bool isFull()
    {
         if (this->top == this->size-1)
            return true;
        return false;

    }

    void display()
    {
        cout<<"[ ";
        for(int i=0;i<=top;i++)
        {
            cout<<data[i]<<", ";
        }
        cout<<"]";
        cout<<endl;
    }

    int seek(int index)
    {
        if(index<0 || index>=size)
        {
            cout<<"\nWrong Index\n";
            return -1;
        }

        return data[index];
    }
};

int main()
{
    Stack s(5);
    s.push(1);
    s.push(4);
    s.push(3);
    s.display();
    cout<<endl<<s.seek(2)<<endl;

}

File: Stack/3_seek.js
class Stack
{
    constructor(size)
    {
        this.size = size;
        this.top  = -1;
        this.data = {};
    }

    
    push(element)
    {
        if(this.isFull())
        {
            console.log("Can not push to already full stack");
            return;
        }

        this.top++;
        this.data[this.top] = element;
    }

    isFull()
    {
         if (this.top == this.size-1)
            return true;
        return false;

    }

    display()
    {
        for(let i=0;i<=this.top;i++)
        {
            console.log(this.data[i]);
        }
    }

    seek(index)
    {
        if(index<0 || index>=this.size)
        {
            console.log("\nWrong Index\n")
            return -1;
        }

        return this.data[index];
    }

}

s = new Stack(5);
s.push(1);
s.push(4);
s.push(3);
s.display();
console.log("seek",s.seek(2))
s.display();